### ISSUE TITLE
Add accessibility tests to PR review

### ISSUE THREAD

**Issue by isabela-pf** [jupyterlab/jupyterlab#9742]
### Problem

Accessibility is a critical aspect of any piece of software, JupyterLab included. In our accessibility meetings, we've briefly discussed the need for automated accessibility testing to catch some accessibility oversights (no test will capture all as far as I know) and to remind people submitting and reviewing PRs that they should be following accessibility guidelines so that
- The team's current efforts to make JupyterLab accessible are not in vain as more code is added
- People can get feedback and gain awareness about accessibility without singularly relying on the feedback of team members that have prior experience
-  The JupyterLab community can grow to prioritize accessibility as a focus rather than a niche concern!
### Proposed Solution

We need to set up automated accessibility tests. What tests make sense, work for JupyterLab's code base, and provide helpful output is still up for discussion. 

Here's a list of possibilities I posted on another issue (sourced from [W3C](https://www.w3.org/WAI/ER/tools/)):
-  [squizlabs / HTML_CodeSniffer](https://github.com/squizlabs/HTML_CodeSniffer)
- IBM's [accessibility-checker](https://www.npmjs.com/package/accessibility-checker)
- [AccessLint / accesslint.js](https://github.com/AccessLint/accesslint.js) 
- [pa11y / pa11y](https://github.com/pa11y/pa11y-ci)
- [dequelabs / axe-core](https://github.com/dequelabs/axe-core)

Carol recommended potential resources at https://accessibility.18f.gov/ too!

From my limited understanding of JupyterLab's code base, we might need to modify existing tests to have helpful outputs since many of these tools are designed for evaluating more standard websites than the kind of web app JupyterLab is.

### Additional context

I've briefly discussed this with community members in nteract community meetings, JupyterLab accessibility meetings, and [JupyterLab Classic #80](https://github.com/jtpio/jupyterlab-classic/issues/80). I'm making this issue now to consolidate discussion where the changes would need to be made.

If it's any help, I know that nteract has a process for this set up that we might be able to follow, but we will have to use different tests since theirs are based on React. I'm also pinging @trallard because she mentioned having experience with some of the above tests and interest in seeing this happen for JupyterLab.

**Comment by bollwyvl** [comment]
@isabela-pf thank you for starting this. 

I have no experience with any other automated tools,  but can verify that `pa11y` is a viable option. Here's a [concrete workflow on pydata-sphinx-theme](https://github.com/pydata/pydata-sphinx-theme/pull/294). 

`pa11y` has support for english-like, selector-based [actions](https://github.com/pa11y/pa11y#actions), which is probably good enough to describe the initial 80% case of:
- open browser
- do a couple things to get a UI up (open a notebook, open the settings, open a terminal, etc.)
- find issues it knows how to find
- report them

Whichever tool is chosen, it should be able to collate the results as JSON/HTML.

Not everything is going to get itself fixed immediately, so having a clear, data-driven roadmap of "we haven't fixed this rule/class of issues today, but want to" seems important to getting started. Driving the tests/reports off this, each _fix this one rule, then keep it fixed_ can be one/a few PRs. Combining the roadmap with reports can then show concrete progress without _too_ much engineering overhead.

If the CI hit _is_ bad, an option would be running this as another repo, e.g. `jupyterlab/accessibility`:
- run nightly against the main branch of this repo
- publish the HTML/JSON reports to its own gh-pages
- once a rule is considered "fixed", start posting regressions as issues on this repo
  - gotta be careful, here: if it's just noise/too many false positives, peoples' eyes start to glaze over 

While this might approach will not catch regressions at the source, per-PR-push, it make it more reasonable to get up and running, and demonstrating results. Eventually, if the approach worked, said repo could offer a github action (or locally-installable package/installer/whatever) that extension authors (or this repo itself) could use to put this concern under more frequent assessment

**Comment by krassowski** [comment]
Just FYI, on jupyter.org we added Lighthouse in https://github.com/jupyter/jupyter.github.io/pull/579. Lighthouse uses a subset of axe (see https://github.com/GoogleChrome/lighthouse/issues/7510).

The problem with Ligthouse is that it is mostly targeted at static pages; it has a [flows](https://github.com/GoogleChrome/lighthouse/blob/de99ab2dc22db435ee6c2501742d7cf6acd11603/docs/user-flows.md) feature and it works with puppeteer; it does not seem to have any special support for playwright (thus Galata), but there is a third-party [playwright-lighthouse](https://github.com/abhinaba-ghosh/playwright-lighthouse) plugin.

If you still have not decided on the tool, I would like to put forward two extra criteria:
- the ability to customise the audits (to disable some bad ones, WCAG 2 contrast checks I am looking at you, https://github.com/jupyter/accessibility/issues/65), and
- the ability to use newer checks from newer WCAG versions: while you will be likely optimizing for WCAG 2.x as the current, wouldn't it be nice to have a little message like:
   - "hey, this fails on WCAG 2, but it's a problem with WCAG 2, in WCAG 3 it will work, no need to change it", or
   - "hey, if you do a little better it will also satisfy the upcoming WCAG 3".
